example/main.go

```diff
@@ -2,152 +2,262 @@ package main
 
 import (
 	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/azure"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/openai"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/vertex"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"time"
 
-	"github.com/Not-Diamond/go-notdiamond/pkg/http/response"
-	"github.com/Not-Diamond/go-notdiamond/pkg/transport"
+	"github.com/joho/godotenv"
+)
 
-	"example/config"
+// AWS Signature V4 constants
+const (
+	awsAlgorithm   = "AWS4-HMAC-SHA256"
+	awsRequestType = "aws4_request"
+	awsService     = "bedrock-runtime"
+	awsContentType = "application/json"
+	awsAccept      = "application/json"
+	awsBedrockHost = "bedrock-runtime.%s.amazonaws.com"
+	awsBedrockPath = "/model/%s/invoke"
 )
 
-func main() {
-	// Load configuration
-	cfg := config.LoadConfig()
+// BedrockConfig holds AWS Bedrock configuration
+type BedrockConfig struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Region          string
+}
+
+// ClaudeRequest represents the request format for Claude models
+type ClaudeRequest struct {
+	AnthropicVersion string                   `json:"anthropic_version"`
+	MaxTokens        int                      `json:"max_tokens"`
+	Messages         []map[string]interface{} `json:"messages"`
+}
+
+// ClaudeResponse represents the response format from Claude models
+type ClaudeResponse struct {
+	Content    []map[string]interface{} `json:"content"`
+	StopReason string                   `json:"stop_reason"`
+	Usage      map[string]int           `json:"usage"`
+}
 
-	// Create API requests
-	openaiRequest, err := openai.NewRequest("https://api.openai.com/v1/chat/completions", cfg.OpenAIAPIKey)
+func main() {
+	// Load .env file
+	err := godotenv.Load(".env")
 	if err != nil {
```

Comment:
When possible, I like to combine these into one line to make it easier to connect the error with the if, and also not make a function-scoped variable. Go has a nice syntax for this you may not know of:

```suggestion
	if err := godotenv.Load(".env"); err != nil {
```